packages/opentelemetry-core/src/context/propagation/HttpTraceContext.ts

```diff
@@ -27,7 +27,7 @@ import { getParentSpanContext, setExtractedSpanContext } from '../context';
 
 export const TRACE_PARENT_HEADER = 'traceparent';
 export const TRACE_STATE_HEADER = 'tracestate';
-const VALID_TRACE_PARENT_REGEX = /^00-([\da-f]{32})-([\da-f]{16})-([\da-f]{2})$/;
+const VALID_TRACE_PARENT_REGEX = /^(?!ff)[\da-f]{2}-([\da-f]{32})-([\da-f]{16})-([\da-f]{2})(-|$)/;
```

Comment:
The change for the version looks good to me. I haven't seen/heard about future field. Could you please elaborate more on that, also share the link, if any.

Think @dyladan should weigh in on this.

Reply:
https://www.w3.org/TR/trace-context/#versioning-of-traceparent

This talks about future versioning. The interesting part is this line "Parse the sampled bit of flags (2 characters from the third dash). Vendors MUST check that the 2 characters are either at the end of the string or followed by a dash."

Also, when you look at the w3c test harness, this test case seems to suggest that future fields are a possibility https://github.com/w3c/trace-context/blob/master/test/test.py#L207

Reply:
Yes, in future versions of the spec, new fields may be added by appending a dash and the new field. This is correct.